### DISCUSSION TITLE
What is the significance of 656e63?

### DISCUSSION THREAD

**Discussion by uhthomas** [cue-lang/cue#2332] [General]
I noticed that json and yaml packages are imported as `json656e63` or `yaml656e63`. Is there some significance behind this or is it just random, to prevent conflict with existing fields which may or may not be called `json` or `yaml`?

**Comment by mvdan** [comment]
When you say you noticed the packages are imported as those names, where do you see that?

**Reply by uhthomas** [reply]
**test.yaml**

```yaml
abc: "please see more info at: https://example.com/"
```
---

```sh
❯ cue import -R test.yaml
```

```cue
import yaml656e63 "encoding/yaml"

abc: yaml656e63.Marshal(_cue_abc)
let _cue_abc = {
	"please see more info at": "https://example.com/"
}
```

**Reply by myitcv** [reply]
My guess (without looking at the code) it's that `656e63` is the first value in a pseudo-random sequence, used as a suffix for the identifier `yaml` to rename the import of `encoding/yaml` so that it cannot possibly clash with other fields.
